### PR TITLE
Logging Improvements

### PR DESCRIPTION
**Pull request by Dala0**
Changes the default log level to INFO.

Adds support for loading the global log level from the <log-level> tag if present. Legal values are CRITICAL, ERROR, WARNING, INFO, and DEBUG. This value is then used to change the log level setting of the main logger object.

Adds support for loading path specific log levels from the <log-level> tag if present for that path. Legal values are CRITICAL, ERROR, WARNING, INFO, and DEBUG. This value is then used to change the log level of unique child logger objects called from the WatchDogBasedSystem and WatchmanBlacklist classes.

Tested on a local virtual machine.

@fredex42

### CODE REVIEW

**Comment by Dala0** [comment]
[Make Watchman Read Log Level Settings from the Config. File](https://trello.com/c/3kCMvwLl/3004-make-watchman-read-log-level-settings-from-the-config-file)

File: watcher/blacklist.py [hunk 1]
```diff
@@ -7,7 +7,12 @@
 
 
 class WatchmanBlacklist(object):
-    def __init__(self, config_xml = None):
+    def __init__(self, config_xml = None, location=None, loglevel=None):
+        self.logger = logging.getLogger('WatchmanBlacklist.{0}'.format(location))
+        try:
+            self.set_log_level(loglevel)
+        except Exception:
+            pass
```
Comment (fredex42) [inline watcher/blacklist.py:17]: would be nice to output a WARNING here if something fails, just so we know

**Review by fredex42 — CHANGES_REQUESTED** [review]
one tiny change, see inline

File: watcher/blacklist.py (hunk 1 above)
Comment (Dala0) [reply watcher/blacklist.py:17]: Done.

**Review by fredex42 — APPROVED** [review]
👍

**Pull request merged**